cdc/log.cc

```diff
@@ -693,9 +714,7 @@ class transformer final {
                                     [&](const user_type_impl& type) {
                                         std::vector<bytes_opt> res(type.size());
                                         std::array<bytes_view_opt, 1> tmp;
-                                        // type of the actual value in log column tuple.
-                                        auto res_type = static_pointer_cast<const tuple_type_impl>(dst->type)->type(1);
-                                        auto tt = static_pointer_cast<const tuple_type_impl>(res_type);
+                                        auto tt = static_pointer_cast<const tuple_type_impl>(dst->type);
```

Comment:
Is the code candling non-frozen UDTs correctly? Shouldn't it create deleted_elements for non-frozen UDTs as well?

Reply:
No. UDT:s cannot use a "deleted_elements", since UDT parts have different types. This is why we have the dreaded `tuple<tuple<typ1>, tuple<typ2>...>`

Reply:
Sorry, noticed this discussion just now. Copy-pasting my comment from below:

Is there no way to use the deleted_elements_ column for UDTs as well?

E.g.:
-    for deleted elements, just use a set of field indexes. We could also use a set of strings (field names) I guess, but that's not resistant to field renames, so field indexes seem better.
-    for elements set to non-null value, use the original UDT type, but frozen.
-    for fields that were not affected, just don't set them in the frozen UDT.